ddpui/api/charts_api.py

```
@@ -0,0 +1,1010 @@
+            # Get the aggregated value using the metric alias
+            metric_alias = metrics[0].alias or f"{metrics[0].aggregation}_{metrics[0].column}"
+            value = row.get(metric_alias)
+
+            if region_name and value is not None:
+                map_data.append({"name": str(region_name), "value": float(value)})
+
+        return {"success": True, "data": map_data, "count": len(map_data)}
+
+    except Exception as e:
+        logger.error(f"Error generating map data overlay: {str(e)}")
+        raise HttpError(500, f"Error generating map data overlay: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)